src/addrdb.cpp

```diff
@@ -161,13 +161,13 @@ bool CAddrDB::Read(CAddrMan& addr, CDataStream& ssPeers)
 void DumpAnchors(const fs::path& anchors_db_path, const std::vector<CAddress>& anchors)
 {
     LOG_TIME_SECONDS(strprintf("Flush %d outbound block-relay-only peer addresses to anchors.dat", anchors.size()));
-    SerializeFileDB("anchors", anchors_db_path, anchors);
+    SerializeFileDB("anchors", anchors_db_path, anchors, CLIENT_VERSION | ADDRV2_FORMAT);
```

Comment:
Here and below in deserialize, why add `CLIENT_VERSION` when it is going to be ignored during ser/deser (they only check `s.GetVersion() & ADDRV2_FORMAT`)?

Reply:
That's just common in the codebase; use CLIENT_VERSION for disk serializations, PROTOCOL_VERSION for network serializations.

I think this whole idea of streams having a version numbers with magic meanings, and implicit (=bad) compatibility properties for anything relying on it are silly, but getting rid of it isn't for this PR.